src/app/login-dialog/login-dialog.component.html

```diff
@@ -0,0 +1,83 @@
+
+<!--Unless required by applicable law or agreed to in writing, software-->
+<!--distributed under the License is distributed on an "AS IS" BASIS,-->
+<!--WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.-->
+<!--See the License for the specific language governing permissions and-->
+<!--limitations under the License.-->
+
+<div class="md-hue-2 login-bg"
+     md-colors="{background: 'default-primary-700'}"
+     fxLayout="column"
+     fxLayoutAlign="center center"
+     flex>
+  <mat-dialog-content>
+    <mat-card align="end">
+      <i class="material-icons close-modal" (click)="close()">
+        close
+      </i>
+    </mat-card>
+    <mat-card fxLayout="column" style="min-width: 650px" layout-padding>
+      <mat-card-content fxLayout="row" fxLayoutGap="10">
+        <mat-card-content fxLayout="column" fxFlex="50">
+          <p>Log in with your social account</p>
+          <div flex></div>
+          <button (click)="github()" class="mat-raised-button github social-button" layout="row" mat-button>
+            <img src="/assets/images/github.svg" class="social-logo">Github
+          </button>
+          <button ng-click="ctrl.signInWithSocial(loginForm, 'google')" class="mat-raised-button google social-button"
+                  layout="row" mat-button>
+            <img src="/assets/images/google-plus.svg" class="social-logo">Google
+          </button>
+          <button ng-click="ctrl.signInWithSocial(loginForm, 'twitter')" class="mat-raised-button twitter social-button"
+                  layout="row" mat-button>
+            <img src="/assets/images/twitter.svg" class="social-logo">Twitter
+          </button>
+          <div flex></div>
+
+        </mat-card-content>
+        <div class="vertical-divider"></div>
+          <mat-card-content layout="column" fxFlex="50" [formGroup]="loginForm" novalidate >
+            <p>Or you can <a href="mailto:niko@biosustain.dtu.dk">contact us</a> and we provide you with credentials</p>
+
+            <div fxLayoutAlign="center center">
+              <mat-icon>person</mat-icon>
+              <mat-form-field fxFlex>
+                <input formControlName="username" placeholder="Username" matInput>
+              </mat-form-field>
+            </div>
+
+            <div fxLayoutAlign="center center">
+              <mat-icon>lock</mat-icon>
+              <mat-form-field fxFlex>
+                <input formControlName="password" type="password" placeholder="Password" matInput>
+              </mat-form-field>
+            </div>
+            <div layout="row" flex [hidden]="uiStatus ==! 'error'">
+              <span fxFlex class="error">{{error}}</span>
+            </div>
+            <div layout="row" flex>
+              <span fxFlex></span>
+              <button
+                type="submit"
+                [disabled]="!loginForm.valid"
+                class="mat-raised-button"
+                color="primary"
+                (click)="submit()"
```

Comment:
What if you'd put the whole section inside a `form` element? I think `ngSubmit` could be used, no?